### PR TITLE
Add support for Windows ARM64

### PR DESCRIPTION
**Pull request by anthony-linaro**
This adds proper support for building for Windows ARM64 devices. using MSVC.

It does actually work right now, but only by complete accident, and with no SIMD enabled - this enables everything properly.

As part of this, I updated the version of sse2neon used to one compatible with MSVC - the particular hash I used is the exact one that Blender uses, which is known working (there were some issues with the base 1.7.0 release).

Commands used:
```
mkdir build
cd build
cmake -G"Ninja" ..
cmake --build .
ctest
```

All tests pass. This is using VS2022, I wouldn't really use VS2019 for ARM64 platforms if I'm honest.

I didn't test any of the GPU things, although GLUT may work, as I added ARM64 support to FreeGLUT a few years ago.

Addresses #1859 (so probably of interest to @num3ric also)

### CODE REVIEW

**Comment by num3ric** [comment]
Love it, thanks for the work! Hopefully the CLA & failing tests can be resolved.

File: share/cmake/modules/install/Installsse2neon.cmake [hunk 1]
```diff
@@ -16,7 +16,7 @@ include(FetchContent)
 set(FETCHCONTENT_BASE_DIR "${CMAKE_BINARY_DIR}/ext/build/sse2neon")
 FetchContent_Declare(sse2neon
   GIT_REPOSITORY https://github.com/DLTcollab/sse2neon.git
-  GIT_TAG        v1.6.0
+  GIT_TAG        227cc413fb2d50b2a10073087be96b59d5364aea
```
Comment (num3ric) [inline share/cmake/modules/install/Installsse2neon.cmake:19]: Any reason you couldn't pick an official version number here?

Comment (anthony-linaro) [reply share/cmake/modules/install/Installsse2neon.cmake:19]: Long story short, we uncovered some issues with 1.7.0 in Blender on various platforms, and needed to update to a non-versioned commit, as they only version sse2neon once a year - the issues may or may not affect OCIO, but this is at least a known working and tested version that Blender uses.

**Comment by anthony-linaro** [comment]
I'll have a dig into those tests and try and figure it out - I don't have those platforms to hand, so will be some guesswork involved.

For the CLA, I'm trying to find out who our internal person is that is responsible for EasyCLA so thay can approve it under our corporate account - I'll find out at some point soon I'm sure

**Comment by anthony-linaro** [comment]
Have fixed a small buildfile issue - I am hoping this may resolve the mac issues (sse2neon accidentally got disabled in some configurations).

I have also fixed some additional issues I encountered while debugging (namely it wasn't actually using sse2neon properly outside of the cmake test, oops).

I have been building with python off, there seems to be some sort of Access Violation Exception, but I'm not sure what it is caused by - for now, the rest of the test suite passes so it should be fine.

**Comment by anthony-linaro** [comment]
CLA now done!

**Comment by num3ric** [comment]
Have you tried (cross-)compiling `-A ARM64` from an x64-based PC? Unfortunately #1859 isn't fully addressed here, since compiling this PR still fails for me unless I add:
```
    if(MSVC)
        set(CMAKE_SYSTEM_PROCESSOR ${MSVC_CXX_ARCHITECTURE_ID})
    endif()
```
If not, I get `OCIO_ARCH_X86 1` and `CpuInfo.cpp` fails per issue.

PR looks good otherwise though!

**Comment by anthony-linaro** [comment]
I had not, mainly on the account of me not having a suitable x64 machine to hand 😄 

I'll get one set up for development and give it a go

**Comment by anthony-linaro** [comment]
Commit pushed, should allow cross-compilation. Tested with the command line:

```
cmake -A arm64 .. -DOCIO_BUILD_PYTHON=OFF
```

On a Windows x64 machine - the binaries produices appear to run correctly on my ARM64 machine

**Review by num3ric — APPROVED** [review]

**Comment by doug-walker** [comment]
@anthony-linaro , thanks for the contribution. Regarding reviews, the project requires a minimum of two reviewers. 

This PR makes significant changes to the build system which are not fully tested by our CI, so we definitely need more people to test. This includes validating that the various Mac builds were not affected.

Are you looking for this to go in OCIO 2.4.x or could it wait until 2.5 next fall?

**Comment by anthony-linaro** [comment]
Thanks for the reply @doug-walker - ideally sooner rather than later, so 2.4.x - this would mean Windows ARM64 would be properly compatible with the VFX Reference Platform 2025, rather than having to wait another year.

VFX reference platform compatibility is something that has been requested by a few external partners, and it would probably be a disappointing answer for me to give them, if I told them they need to wait another year, or maintain an out-of-tree patch

**Comment by num3ric** [comment]
+1 on prioritization, to get rid of patches on our next engine release.

Also confirming on my end that this PR successfully resolves #1859.

File: src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp [hunk 2]
```diff
@@ -1844,7 +1844,7 @@ __m128 Renderer_LIN_TO_PQ_SSE<true>::myPower(__m128 x, __m128 exp)
     return ssePower(x, exp);
 }
 
-#ifdef _WIN32
+#if (_MSC_VER >= 1920) && (OCIO_USE_AVX)
```
Comment (lazka) [inline src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp:1847]: nitpick: `// _WIN32` below not updated

**Review by cozdas — CHANGES_REQUESTED** [review]
Thanks for extending the Windows support for Arm64.

When I compile this branch with Visual Studio (as opposed to ninja generator), I get compiler errors in CPUinfo.cpp for _xgetbv and __cpuid x86 intrinsics which are guarded only with _MSC_VER macro thus active for ARM targets as well. Can you please make sure that this compiles with Windows native generator as well.

**Comment by doug-walker** [comment]
@anthony-linaro , @num3ric , as a reminder, our 2.4.1 release is going out on Wednesday.  I know you indicated you wanted this PR included, but unfortunately we don't feel that it is ready to merge.

**Comment by cozdas** [comment]
BTW @anthony-linaro  I tested the branch with Windows 11 running in a parallels VM on macbook pro M2 and I'm not super experienced with Arm based Windows development. Let me know if you think that my test environment is not ideal or if you think that I'm overlooking something.

**Comment by Mushroom** [comment]
@doug-walker That's disappointing to hear, with a more expeditious review, I could have worked through the issues, but understandably circumstances can prevent this. Will there be another 2.4.x release?

@cozdas I can't repro this, which version of VS are you using? I used these build instructions to use the VS generator:
```
cmake .. -DOCIO_BUILD_PYTHON=OFF
cmake --build . -- config Release
```

I also tried without the second line, opening the produiced sln file in VS2022, choosing "Release", and building like that, still with no issues.

File: src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp (hunk 2 above)
Comment (Mushroom) [reply src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp:1847]: Fixed, thanks for the review!

**Comment by Mushroom** [comment]
Doug - That's disappointing to hear, with a more expeditious review, I could have worked through the issues, but understandably circumstances can prevent this. Will there be another 2.4.x release?

Cozdas - I can't repro this, which version of VS are you using? I used these build instructions to use the VS generator:
```
cmake .. -DOCIO_BUILD_PYTHON=OFF
cmake --build . --config Release
```

I also tried without the second line, opening the produiced sln file in VS2022, choosing "Release", and building like that, still with no issues.

**Comment by num3ric** [comment]
@cozdas I did have the same compilation errors in visual studio before 04a723a. Are you sure you're on the latest version of this PR?

**Comment by cozdas** [comment]
Since it worked for you, I re-tried after updating the toolchain, visual studio and cmake to the latest version and re-tried and it compiled on my end too. Basic core tests work too.

**Review by cozdas — APPROVED** [review]

File: share/cmake/utils/CompilerFlags.cmake [hunk 3]
```diff
@@ -18,7 +18,12 @@ if(OCIO_USE_SIMD)
 
     if (OCIO_USE_SSE2NEON AND COMPILER_SUPPORTS_ARM_NEON)
         include(CheckSupportSSEUsingSSE2NEON)
-        if(NOT COMPILER_SUPPORTS_SSE_WITH_SSE2NEON)
+        if(COMPILER_SUPPORTS_SSE_WITH_SSE2NEON)
+            if(WIN32)
```
[the remark this answers is not in the document]
Comment (anthony-linaro) [reply share/cmake/utils/CompilerFlags.cmake:22]: Good catch - I'd forgotten about GNU-style on Windows, I checked clang, but I was usng clang-cl. Will fix shortly.

File: src/OpenColorIO/CPUInfo.cpp [hunk 4]
```diff
@@ -183,7 +183,7 @@ CPUInfo::CPUInfo()
     }
 }
 
-#elif defined(__aarch64__) // ARM Processor or Apple ARM.
+#elif defined(__aarch64__) || defined(_M_ARM64) // ARM Processor or Apple ARM.
```
[the remark this answers is not in the document]
Comment (anthony-linaro) [reply src/OpenColorIO/CPUInfo.cpp:186]: I'm not sure I agree - this comment is a continuation of the comment from the initial `#if` directive, and explains what we are checking for in each case. If another architecture was added in the future, it makes sense to explain what each section checks for.

[the remark this answers is not in the document]
Comment (anthony-linaro) [reply src/OpenColorIO/CPUInfo.cpp:186]: Most other locations of `_M_ARM64` do have a comment: [1](https://github.com/anthony-linaro/OpenColorIO/blob/add-arm64-support/src/OpenColorIO/CPUInfoConfig.h.in#L9), [2](https://github.com/anthony-linaro/OpenColorIO/blob/add-arm64-support/src/OpenColorIO/CPUInfoConfig.h.in#L26), [3](https://github.com/anthony-linaro/OpenColorIO/blob/add-arm64-support/src/OpenColorIO/SSE.h#L16), [4](https://github.com/anthony-linaro/OpenColorIO/blob/add-arm64-support/src/OpenColorIO/SSE.h#L93), [5](https://github.com/anthony-linaro/OpenColorIO/blob/add-arm64-support/src/OpenColorIO/SSE2.h#L14), for example - however, if you feel strongly about this, I'll remove it (or modify the comment to fix the ambiguity you descirbe), it's only a comment, after all :)

[the remark this answers is not in the document]
Comment (anthony-linaro) [reply src/OpenColorIO/CPUInfo.cpp]: Fair enough :) I have updated the comment, hopefully this solves the issue!

**Comment by anthony-linaro** [comment]
Do we know who else needs to review this for it to get merged?

**Review by doug-walker — APPROVED** [review]
Thanks for the work on this. I think the best approach at this point is to merge it into the main branch, which will get more people testing it. 

If no problems are encountered over the next few months, we will include it in OCIO 2.4.2.

**Pull request merged**